clang/unittests/AST/ASTImporterTest.cpp

```diff
@@ -8147,6 +8149,26 @@ TEST_P(ImportFunctions, CTADWithLocalTypedef) {
   ASSERT_TRUE(ToD);
 }
 
+TEST_P(ImportFunctions, CTADAliasTemplate) {
+  Decl *TU = getTuDecl(
+      R"(
+      template <typename T> struct A {
+        A(T);
+      };
+      template<typename T>
+      using B = A<T>;
+      B b{(int)0};
+      )",
+      Lang_CXX20, "input.cc");
+  auto *FromD = FirstDeclMatcher<CXXDeductionGuideDecl>().match(
+      TU, cxxDeductionGuideDecl(hasParameter(0, hasType(asString("int")))));
+  auto *ToD = Import(FromD, Lang_CXX20);
+  ASSERT_TRUE(ToD);
+  EXPECT_TRUE(ToD->getSourceDeductionGuideKind() ==
+              CXXDeductionGuideDecl::SourceDeductionGuideKind::Alias);
```

Comment:
Can we also get a test that covers the `None` case to ensure we are serializing and deserializing correctly for both.

Testing should if possible always cover all path, otherwise this opens us to undetected regressions in the untested paths.

Reply:
Done, I was already testing `getSourceDeductionGuide` for the `nulltptr` case so I have now added assertions for `getSourceDeductionGuideKind` being `None`.